traffic_ops/traffic_ops_golang/acme/acme_account.go

```diff
@@ -0,0 +1,261 @@
+package acme
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import (
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/apache/trafficcontrol/lib/go-tc"
+	"github.com/apache/trafficcontrol/lib/go-tc/tovalidate"
+	"github.com/apache/trafficcontrol/lib/go-util"
+	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/api"
+	"github.com/go-ozzo/ozzo-validation"
+	"net/http"
+)
+
+type AcmeAccount struct {
+	Email      *string `json:"email" db:"email"`
+	PrivateKey *string `json:"private_key" db:"private_key"`
+	Uri        *string `json:"uri" db:"uri"`
+	Provider   *string `json:"provider" db:"provider"`
+}
+
+func (aa *AcmeAccount) Validate(tx *sql.Tx) error {
+
+	errs := validation.Errors{
+		"email":       validation.Validate(aa.Email, validation.Required),
+		"private_key": validation.Validate(aa.PrivateKey, validation.Required),
+		"uri":         validation.Validate(aa.Uri, validation.Required),
+		"provider":    validation.Validate(aa.Provider, validation.Required),
+	}
+
+	return util.JoinErrs(tovalidate.ToErrors(errs))
+}
+
+func (aa *AcmeAccount) ValidateUpdate(tx *sql.Tx) error {
+
+	errs := validation.Errors{
+		"email":    validation.Validate(aa.Email, validation.Required),
+		"provider": validation.Validate(aa.Provider, validation.Required),
+	}
+
+	return util.JoinErrs(tovalidate.ToErrors(errs))
+}
+
+const readQuery = `SELECT email, private_key, uri, provider FROM acme_account`
+const createQuery = `INSERT INTO acme_account (email, private_key, uri, provider) VALUES (:email, :private_key, :uri, :provider) RETURNING email, provider`
+const updateQuery = `UPDATE acme_account SET private_key=:private_key, uri=:uri WHERE email=:email and provider=:provider RETURNING email, provider`
+const deleteQuery = `DELETE FROM acme_account WHERE email=$1 and provider=$2`
+
+func Read(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	tx := inf.Tx.Tx
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	var acmeAccounts []AcmeAccount
```

Comment:
The zero-value of a slice is `nil`, so if you have no accounts stored, I think a GET request here would return
```json
{ "response": null }
```
when really we'd want
```json
{ "response": [] }
```